libs/utils/trace.py

```diff
@@ -886,6 +886,69 @@ def _compute_steps_duration(self, df):
 
         return df
 
+    def _windowify_steps_df(self, df, (start, end), len_column='len'):
+        """
+        Slice a dataframe to [start:end], and work on the time data so that it
+        makes sense within the interval.
```

Comment:
This functionality is so tight related to the new `_compute_steps_duration` which I'm wondering if it does not make more sense to fold this code into that call. We can have a default `(start, end) = (0, None)` which does not do any trimming at all... otherwise, you just add this functionality at then 

Based on previous patch comment, the new prototype would become:

```python
def addEventsDeltas(self, df, col_name='delta', trim=(0, None))
```

"Slice a dataframe using the specified [start:end] timestamps, and fix col_name values to match the new time boundaries"?